scheduler/core/components/optimizer/greedymax.py

```diff
@@ -15,18 +15,45 @@
 from .base import BaseOptimizer
 
 import numpy as np
-import astropy.units as u
+# import astropy.units as u
 
 
 class GreedyMaxOptimizer(BaseOptimizer):
     """
     GreedyMax is an optimizer that schedules the visits for the rest of the night in a greedy fashion.
     """
 
-    def __init__(self, seed=42):
+    def __init__(self):
         self.groups = []
-        self.groupids = []
+        self.group_ids = []
+        # self.obs_groups = []     # remove if not used
+        self.obs_group_ids = []
         self.timelines = []
+        self.sites = []
+        self.min_visit_len = timedelta(minutes=30)
```

Comment:
Yes, will do this. What is the convention about when a variable is private?

Reply:
If the variable doesn't need to be used outside of the class it's in, it should be marked private, which means its name starts with a single underscore. You can access it inside the class as much as you want without warnings, of course, even in a static context like this.

Note that Python won't explicitly prohibit you from using it outside of the class: most IDEs will just generate a warning. It can be useful or necessary to occasionally access it outside the class during production or when testing.

Python will obfuscate a variable entirely if it starts with a double underscore, but this isn't recommended, and it still doesn't prevent you from using the variable outside of the class since the obfuscation is predictable: If you have a class called `MyClass` and you have `__my_var`, and say an instance `my_class` of `MyClass`, you can still get the value by accessing:
```python
my_class._MyClass__my_var
```

So there is no real enforcement. It's just convenient to have the IDE warn you when accessing private variables and it's generally understood amongst Python developers that you shouldn't change a private variable's value outside of the class.

Reply:
Actually, I do want to make min_visit_length a passable parameter. So, should that then be defined with something like?
    def __init__(self, min_visit_len = timedelta(minutes=30)):
Then, how I pass this from upstream? This is for greedymax only so should not affect other optimizers.

Reply:
To be able to define it and include a default value, I'd recommend doing:
```python
def __init__(self, min_visit_length = timedelta(minutes=30)):
    ...
    self._min_visit_length = min_visit_length
    ...
```
Then you can pass it to GreedyMax as a parameter with the default value, and it's stored as a private variable inside GreedyMax, where you refer to it as `self._min_visit_length`,